src/accordionpane/AccordionPane.ts

```diff
@@ -0,0 +1,71 @@
+import { after } from '@dojo/core/aspect';
+import { assign } from '@dojo/core/lang';
+import { DNode, WNode } from '@dojo/widget-core/interfaces';
+import { includes } from '@dojo/shim/array';
+import { ThemeableMixin, ThemeableProperties, theme } from '@dojo/widget-core/mixins/Themeable';
+import { v } from '@dojo/widget-core/d';
+import { WidgetBase } from '@dojo/widget-core/WidgetBase';
+
+import TitlePane from '../titlepane/TitlePane';
+
+import * as css from './styles/accordionPane.m.css';
+
+/**
+ * @type AccordionPaneProperties
+ *
+ * Properties that can be set on AccordionPane components
+ *
+ * @property onRequestClose   Called when the title of an open pane is clicked
+ * @property onRequestOpen    Called when the title of a closed pane is clicked
+ * @property openKeys         Array of TitlePane keys indicating which panes should be open
+ */
+export interface AccordionPaneProperties extends ThemeableProperties {
+	onRequestClose?(key: string): void;
+	onRequestOpen?(key: string): void;
+	openKeys?: string[];
+};
+
+export const AccordionPaneBase = ThemeableMixin(WidgetBase);
+
+@theme(css)
+export default class AccordionPane extends AccordionPaneBase<AccordionPaneProperties, WNode<TitlePane>> {
+	private _assignCallback(child: WNode<TitlePane>, functionName: string, callback: (key: string) => void) {
+		const existingProperty = (child.properties as any)[functionName];
+		const property = () => { callback.call(this, `${ child.properties.key }`); };
+
+		return existingProperty ? after(existingProperty, property) : property;
+	}
+
+	protected onRequestClose(key: string) {
+		const { onRequestClose } = this.properties;
+		onRequestClose && onRequestClose(key);
+	}
+
+	protected onRequestOpen(key: string) {
+		const { onRequestOpen } = this.properties;
+		onRequestOpen && onRequestOpen(key);
+	}
+
+	protected renderChildren(): DNode[] {
+		const {
+			openKeys = [],
+			theme
+		} = this.properties;
+
+		return this.children.filter(child => child).map(child => {
+			// null checks skipped since children are filtered prior to mapping
+			assign(child!.properties, {
+				onRequestClose: this._assignCallback(child!, 'onRequestClose', this.onRequestClose),
+				onRequestOpen: this._assignCallback(child!, 'onRequestOpen', this.onRequestOpen),
+				open: includes(openKeys, child!.properties.key),
+				theme
+			});
+
+			return child;
+		});
+	}
+
+	render(): DNode {
+		return v('div', { classes: this.classes(css.root) }, this.renderChildren());
```

Comment:
You know what would be really cool? If we added arrow key navigation through panes 😁 
(am I evil? I might be evil)

Reply:
We already support tabbing between panes and using enter to open and close them, so I figured that's sufficient for now until our Ultimate a11y Overhaul Extravaganza.

Reply:
I'll be OK with it if you open a separate issue for adding arrow key navigation :D

Reply:
Fair, thought we had one. #342